src/components/badge/badge.test.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import Badge from "./badge.component";
+
+const renderComponent = (props = {}) =>
+  render(<Badge {...props}>Button</Badge>);
+
+describe("Badge", () => {
+  it("should render correctly if counter is a number", async () => {
+    renderComponent({ counter: 1 });
+
+    expect(screen.getByText("1")).toBeVisible();
+  });
+
+  it("should render correctly if counter is a string", async () => {
+    renderComponent({ counter: "1" });
+
+    expect(screen.getByText("1")).toBeVisible();
+  });
+
+  it("should render number when counter is between 1 and 99", async () => {
+    renderComponent({ counter: 50 });
+
+    expect(screen.getByText("50")).toBeVisible();
+  });
+
+  it("should render number `99` when counter is higher than 99", async () => {
+    renderComponent({ counter: 101 });
+
+    expect(screen.getByText("99")).toBeVisible();
+  });
+
+  it("should not render badge when counter is `0`", async () => {
+    renderComponent({ counter: 0, onClick: () => {} });
+
+    expect(screen.queryByText("0")).not.toBeInTheDocument();
+    expect(screen.queryByRole("button")).not.toBeInTheDocument();
+  });
+
+  it("should not render badge when counter is not set", async () => {
+    renderComponent({ onClick: () => {} });
+
+    expect(screen.queryByRole("button")).not.toBeInTheDocument();
+  });
+
+  it("should render as a button element when onClick is set", async () => {
+    renderComponent({ counter: 9, onClick: () => {} });
+
+    // investigate why toBeVisible() fails to find "9"
+    expect(screen.getByText("9")).toBeInTheDocument();
+    expect(screen.getByRole("button")).toBeVisible();
+  });
+
+  it("should not render as a button if onClick is not set", async () => {
+    renderComponent({ counter: 9 });
+
+    expect(screen.getByText("9")).toBeVisible();
+    expect(screen.queryByRole("button")).not.toBeInTheDocument();
+  });
+
+  it("should have the relevant aria-label when aria-label is specified", async () => {
+    renderComponent({ counter: 9, "aria-label": "Generic aria message" });
+
+    expect(screen.getByLabelText("Generic aria message")).toBeVisible();
+  });
+
+  describe("Styles", () => {
```

Comment:
suggestion: I think we should avoid nesting the `describe` here, the `it` blocks have enough of a description imo